src/scripts/main.js

```diff
@@ -1,3 +1,129 @@
 'use strict';
 
-// write your code here
+function keyListener(e) {
+  if (game.getStatus() !== 'playing') {
+    return;
+  }
+
+  switch (e.key) {
+    case 'ArrowDown':
+      game.moveDown();
+      break;
+    case 'ArrowLeft':
+      game.moveLeft();
+      break;
+    case 'ArrowRight':
+      game.moveRight();
+      break;
+    case 'ArrowUp':
+      game.moveUp();
+      break;
+  }
+
+  update();
+}
+
+function update() {
+  game.getState().forEach((row, i) => {
+    row.forEach((el, j) => {
+      let cellCssClass = 'field-cell';
+
+      cellCssClass += el >= 2 ? ` field-cell--${el}` : '';
+      gameCells[i][j].innerText = el || '';
+      gameCells[i][j].className = cellCssClass;
+    });
+  });
+
+  scoreElement.innerText = game.getScore();
+
+  switch (game.getStatus().toUpperCase()) {
+    case 'WIN':
+      changeMessage('WIN');
+      break;
+    case 'LOSE':
+      changeMessage('LOSE');
+      break;
+    case 'NO_MOVES':
+      changeMessage('NO_MOVES');
+      break;
+    default:
+      hideAllMessages();
+      break;
+  }
+}
+
+function start() {
+  switch (game.getStatus()) {
+    case 'idle':
+      game.start();
+      toggleButton('restart');
+      removeStartMessage();
+      break;
+    case 'playing':
+    case 'win':
+    case 'lose':
+      game.restart();
+      toggleButton('start');
+      changeMessage('START');
+      break;
+  }
+
+  update();
+}
+
+function removeStartMessage() {
+  const startMessage = document.querySelector('.message-start');
+
+  if (startMessage) {
+    startMessage.remove();
+  }
+}
+
+function toggleButton(text) {
+  button.classList.toggle('start');
+  button.classList.toggle('restart');
+  button.innerText = text.toUpperCase();
+}
+
+function changeMessage(type) {
+  let messageClass;
+
+  switch (type.toUpperCase()) {
+    case 'WIN':
+      messageClass = '.message-win';
+      break;
+    case 'LOSE':
+      messageClass = '.message-lose';
+      break;
+    case 'START':
+      messageClass = '.message-start';
+      break;
+    case 'NO_MOVES':
+      messageClass = '.message-no-moves';
+      break;
+    default:
+      throw new Error('unknown type');
```

Comment:
The 'changeMessage' function throws an error for an unknown type, which could potentially disrupt the game flow. It would be safer to handle an unknown type gracefully, such as by logging a warning or providing a default behavior.